core/unix/signal_linux_aarch64.c

```diff
@@ -102,43 +102,81 @@ dump_sigcontext(dcontext_t *dcontext, sigcontext_t *sc)
+                 * context handling, these are the simplest and most consistent
+                 * units for calculating sizes and offsets in order to access
+                 * scalable register state in memory.
+                 */
+                LOG(THREAD, LOG_ASYNCH, 2, "\tQuadwords (128 bits) per vector %d\n\n",
+                    quads_per_vector);
                 LOG(THREAD, LOG_ASYNCH, 2, "\tSVE_SIG_ZREG_SIZE %d\n",
-                    SVE_SIG_ZREG_SIZE(vq));
+                    SVE_SIG_ZREG_SIZE(quads_per_vector));
                 LOG(THREAD, LOG_ASYNCH, 2, "\tSVE_SIG_PREG_SIZE %d\n",
-                    SVE_SIG_PREG_SIZE(vq));
+                    SVE_SIG_PREG_SIZE(quads_per_vector));
                 LOG(THREAD, LOG_ASYNCH, 2, "\tSVE_SIG_FFR_SIZE  %d\n",
-                    SVE_SIG_FFR_SIZE(vq));
+                    SVE_SIG_FFR_SIZE(quads_per_vector));
                 LOG(THREAD, LOG_ASYNCH, 2, "\tsve->head.size %d\n\n", sve->head.size);
                 LOG(THREAD, LOG_ASYNCH, 2, "\tSVE_SIG_ZREGS_OFFSET %d\n",
                     SVE_SIG_ZREGS_OFFSET);
                 LOG(THREAD, LOG_ASYNCH, 2, "\tSVE_SIG_PREGS_OFFSET %d\n",
-                    SVE_SIG_PREGS_OFFSET(vq));
+                    SVE_SIG_PREGS_OFFSET(quads_per_vector));
                 LOG(THREAD, LOG_ASYNCH, 2, "\tSVE_SIG_FFR_OFFSET   %d\n\n",
-                    SVE_SIG_FFR_OFFSET(vq));
+                    SVE_SIG_FFR_OFFSET(quads_per_vector));
 
-                uint64 vdw; /* A vector's doubleword. */
-                int boff;   /* Byte offset for each doubleword in a vector. */
+                dr_simd_t z;
+                int boff; /* Byte offset for each doubleword in a vector. */
                 for (i = 0; i < MCXT_NUM_SIMD_SVE_SLOTS; i++) {
                     LOG(THREAD, LOG_ASYNCH, 2, "\tz%-2d  0x", i);
-                    for (boff = ((vq * 2) - 1); boff >= 0; boff--) {
-                        vdw = *((uint64 *)((((byte *)sve) + (SVE_SIG_ZREG_OFFSET(vq, i)) +
-                                            (boff * 8))));
-                        LOG(THREAD, LOG_ASYNCH, 2, "%016lx ", vdw);
+                    for (boff = ((quads_per_vector * 2) - 1); boff >= 0; boff--) {
+                        /* We access data in the scalable vector using the
+                         * kernel's SVE_SIG_ZREG_OFFSET macro which gives the
+                         * byte offset into a vector based on units of 128 bits
+                         * (quadwords). In this loop we offset from the start
+                         * of struct sve_context. We log the data as 64 bit
+                         * ints, so 2 per quadword.
+                         *
+                         * For example, for a 256 bit vector (2 quadwords, 4
+                         * doublewords), the byte offset (boff) for each
+                         * scalable vector register is:
+                         * boff=3  vdw=sve+SVE_SIG_ZREG_OFFSET+24
```

Comment:
Please explain "vdw".

Reply:
Good spot! It's (very) short for 'vector doubleword'. Replaced using Z registers' double element to make it clearer.